ambari-server/src/main/java/org/apache/ambari/server/checks/DatabaseConsistencyCheckHelper.java

```diff
@@ -1190,6 +1189,64 @@ static void checkServiceConfigs()  {
 
   }
 
+  static Map<Long, ConfigGroup> collectConfigGroupsWithoutServiceName() {
+    Map<Long, ConfigGroup> configGroupMap = new HashMap<>();
+    Clusters clusters = injector.getInstance(Clusters.class);
+    Map<String, Cluster> clusterMap = clusters.getClusters();
+
+    if (MapUtils.isEmpty(clusterMap))
+      return configGroupMap;
+
+    for (Cluster cluster : clusterMap.values()) {
+      Map<Long, ConfigGroup> configGroups = cluster.getConfigGroups();
+
+      if (MapUtils.isEmpty(configGroups))
+        continue;
+
+      for (ConfigGroup configGroup : configGroups.values()) {
+        if (StringUtils.isEmpty(configGroup.getServiceName())) {
+          configGroupMap.put(configGroup.getId(), configGroup);
+        }
+      }
+    }
+
+    return configGroupMap;
+  }
+
+  static void checkConfigGroupsHasServiceName() {
+    Map<Long, ConfigGroup> configGroupMap = collectConfigGroupsWithoutServiceName();
+    if (MapUtils.isEmpty(configGroupMap))
+      return;
+    StringBuilder output = new StringBuilder("[(ConfigGroup) => ");
+
+    for (ConfigGroup configGroup : configGroupMap.values()) {
+      configGroupMap.put(configGroup.getId(), configGroup);
+      output.append("( ");
+      output.append(configGroup.getName());
+      output.append(" ), ");
+    }
+
+    output.replace(output.lastIndexOf(","), output.length(), "]");
+    warning("You have config groups present in the database with no " +
+            " service name, {}. Run --auto-fix-database to fix " +
+            "this automatically.", output.toString());
+  }
+
+  @Transactional
+  static void fixConfigGroupServiceNames() {
+    Map<Long, ConfigGroup> configGroupMap = collectConfigGroupsWithoutServiceName();
+
+    if (!MapUtils.isEmpty(configGroupMap)) {
+      for (Map.Entry<Long, ConfigGroup> configGroupEntry : configGroupMap.entrySet()) {
+        Long id = configGroupEntry.getKey();
+        ConfigGroup configGroup = configGroupEntry.getValue();
+        LOG.info("Setting service name of config group {} with id {} to {}",
+                configGroup.getName(), id, configGroup.getTag());
+        configGroup.setServiceName(configGroup.getTag());
```

Comment:
Are you sure that `configGroup.getTag()` is always the service name? That's not enforced anywhere, so I'm not so sure we can make this assumption.